File: TP/TP1/Code/5I.cpp
/* Afficher un damier de taille N * M, USER choisi le caractère */
#include <iostream>

using namespace std;

int main()
{
  int n;
  int m;
  char c;
  
  cout << "Quelle est la longueur de la ligne ?\n";
  cin >> n;
	cout << "Quelle est la hauteur du rectangle ?\n";
  cin >> m;
  cout << "Quelle est le caractère souhaité ?\n";
  cin >> c;
  
  for (int i = 0; i < m; ++i) {
    for (int j = 0; j < n; ++j) {
			if ((i % 2 == 0 && j % 2 == 0) || (i % 2 != 0 && j % 2 != 0))
        cout << c;
      else
        cout << " ";
    }
    cout << endl;
  }

  return 0;
}

File: TP/TP1/Code/Python/5I.py
#!/usr/bin/python
# Afficher un damier de taille N * M, USER choisi le caractère
print('Donne moi la hauteur : ', end='')
foo = int(input())
print('Donne moi la largeur : ', end='')
bar = int(input())
print('Donne moi le caractère : ', end='')
baz = str(input())

for i in range(foo):
    for j in range(bar):
        if (i % 2 == 0 and j % 2 == 0) or (i % 2 != 0 and j % 2 != 0):
            print(baz, end='')
        else:
            print(' ', end='')
    print()  # Ici on affiche un saut de ligne
